### ISSUE TITLE
Gradient legend items from ESRI Services include a gap

### ISSUE THREAD

**Issue by richardbarad** [wri/gfw-mapbuilder#273]
It looks like legend items which come from an ESRI service and use a gradient are broken up into multiple images. For an example, see the Potential for Increase in Above-ground Carbon Sequestration where Maximum Tree Cover in Cultivated Areas is Capped at 20% layer in the 3rd group layer in the India atlas below.

https://india.restorationatlas.org/atlas

It looks like there is 20x 20 px image, which is followed by a gap and then a longer strip. Is it possible to have continuous strip for gradient legends from ESRI Services which would be similar to what we have for the API legends.

![image](https://user-images.githubusercontent.com/40440743/48633490-6e02b100-e991-11e8-94d9-60b4a5796f36.png)

**Comment by nbumbarger** [comment]
I'm guessing the Esri legend is configured with gradient breaks like this, to create the illusion of a gap:
```
0%: blue
20%: blue
20.1%: white
25%: white
25.1%: cyan
100%: green
```
The proposed gradient API assumes one color per item and evenly spaces the colors, so this isn't possible. For that effect, I think we would drop the concept of discrete categories and define it like this:
```
{
  labels: ['low', 'high'],
  colors: [{
    percentage: 0,
    color: '#0000ff
  }, {
    percentage: 25,
    color: '#00ff00
  }, {
    percentage: 50,
    color: '#00ffdd
  }, ...]
}
```

**Comment by richardbarad** [comment]
Hi @nbumbarger,

Following up on our discussion yesterday the issue was about the gap which was coming from the ArcGIS Online legend and not the API legend. I just tried looking at the map in the webmap viewer and it seems like the gaps are also present. Is there a way to remove the gap from mapbuilder legends or is this something we need to ask ESRI to fix? If there is not an easy fix on the mapbuilder side we can disregard this task.

Here is the webmap: https://arcg.is/09uDa1

![2018-12-07 17_37_38-window](https://user-images.githubusercontent.com/40440743/49676377-82454580-fa47-11e8-86af-9ab63025338b.png)

**Comment by csphang** [comment]
@richardbarad Right now we are inheriting this from the service. There doesn't seem to be an easy fix to style these to reduce the gap. Here's a link to the service legend for reference: https://gis.forest-atlas.org/server/rest/services/ind/RO_TC_Carbon_final/MapServer/legend

![2018-12-10_15-29-09](https://user-images.githubusercontent.com/13664037/49759702-6aaccd80-fc90-11e8-917a-deb961a6ac7c.png)

**Comment by richardbarad** [comment]
ok. Thanks @csphang. We can put this task on hold / move it to a backlog.

**Comment by richardbarad** [comment]
@csphang & @SwampGuzzler. Our India team brought this up again and I investigated it a bit. Documenting what I found!

It looks like this issue is caused be the height of the `class="legend-row"` when using gradient legends. Currently, the height is set to `24px` which is larger than the height of the image which is `20px`. If I change the height to 20px the gap goes away. 

**24px Height:**

![20px](https://user-images.githubusercontent.com/40440743/67020267-ec992e80-f0cb-11e9-98bf-7185fd859b41.png)

**20px Height:**

![20px](https://user-images.githubusercontent.com/40440743/67020351-0fc3de00-f0cc-11e9-8d68-9e5607dd7dc6.png)
